### PR TITLE
Add threshold below which computation time is not reported.

### PR DESCRIPTION
**Pull request by anntzer**
See #221.

### CODE REVIEW

File: sphinx_gallery/gen_rst.py [hunk 1]
```diff
@@ -638,9 +638,10 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
     save_notebook(example_nb, example_file.replace('.py', '.ipynb'))
     with codecs.open(os.path.join(target_dir, base_image_name + '.rst'),
                      mode='w', encoding='utf-8') as f:
-        example_rst += "**Total running time of the script:**" \
-                       " ({0: .0f} minutes {1: .3f} seconds)\n\n".format(
-                           time_m, time_s)
+        if time_elapsed >= gallery_conf["min_reported_time"]:
```
Comment (choldgraf) [inline sphinx_gallery/gen_rst.py]: is there any situation in which not forcing this to be a float will bite us?

File: doc/advanced_configuration.rst [hunk 2]
```diff
@@ -554,4 +555,18 @@ in your own site's CSS file:
     }
 
 
+.. _min_reported_time:
```
Comment (choldgraf) [inline doc/advanced_configuration.rst]: mind if I clean up this wording a small amount before merging? happy to do it myself as a quick commit-then-merge

Comment (anntzer) [reply doc/advanced_configuration.rst]: Feel free to force push to my branch.

File: sphinx_gallery/gen_rst.py (hunk 1 above)
Comment (anntzer) [reply sphinx_gallery/gen_rst.py]: If somebody sets the min computation time to not a float then that will crash but what can you do?...  Obviously an explicit type check could be added, feel free to push one.

Comment (larsoner) [reply sphinx_gallery/gen_rst.py]: I'd assume YAGNI and rely on the Python error message to be explicit enough

Comment (choldgraf) [reply sphinx_gallery/gen_rst.py]: ah I was just thinking to add a `float(gallery_conf["min_reported_time"])` in there as opposed to doing a check

Comment (larsoner) [reply sphinx_gallery/gen_rst.py]: sure, that might give better errors in any case

Comment (anntzer) [reply sphinx_gallery/gen_rst.py]: That (so support strings when we want a float) I think would be unpythonically wrong.

Comment (larsoner) [reply sphinx_gallery/gen_rst.py]: If we eventually add it as a command-line option I think it would come in as `str`, though

Comment (choldgraf) [reply sphinx_gallery/gen_rst.py]: ok I'm gonna wrap it in float for now and we can make it error later if we want

File: CHANGES.rst [hunk 3]
```diff
@@ -4,6 +4,11 @@ Change Log
 git master
 ----------
 
+New features
+''''''''''''
+
+* Added ``min_reported_time`` configurable.
```
Comment (lesteve) [inline CHANGES.rst]: You will need to shortly explain what this does.

File: doc/advanced_configuration.rst [hunk 4]
```diff
@@ -554,4 +555,18 @@ in your own site's CSS file:
...
+
+Minimal reported time
+=====================
+
+By default, Sphinx-gallery logs and embeds in the html output the time it took
+to run each script.  If the majority of your examples runs quickly, you may not
+need this information.
+
+The ``min_reported_time`` confirguration can be set to a number of seconds.
```
Comment (lesteve) [inline doc/advanced_configuration.rst]: typo in configuration

File: sphinx_gallery/gen_gallery.py [hunk 5]
```diff
@@ -247,11 +248,12 @@ def generate_gallery_rst(app):
 
             fhindex.write(SPHX_GLR_SIG)
 
-    if gallery_conf['plot_gallery']:
+    if gallery_conf['plot_gallery'] and gallery_conf['min_reported_time']:
```
Comment (lesteve) [inline sphinx_gallery/gen_gallery.py]: I don't understand why you are adding `and gallery_conf['min_reported_time']`. The vaiue of min_reported_time is 0 so this does not log by default, right?

File: CHANGES.rst (hunk 3 above)
Comment (anntzer) [reply CHANGES.rst]: done

File: doc/advanced_configuration.rst (hunk 4 above)
Comment (anntzer) [reply doc/advanced_configuration.rst]: fixed

File: sphinx_gallery/gen_gallery.py (hunk 5 above)
Comment (anntzer) [reply sphinx_gallery/gen_gallery.py]: Good catch, probably leftover from earlier design.

**Comment by choldgraf** [comment]
Is this ready to go @anntzer ?

**Comment by anntzer** [comment]
sure.

**Pull request merged**